### DISCUSSION TITLE
Error in defining custom function for electrolyte diffusivity

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#2805] [Q&A] [answered]
import pybamm as pb
import numpy as np
import pandas as pd

pb.set_logging_level("NOTICE")
model = pb.lithium_ion.DFN()
param = pb.ParameterValues("Chen2020")

// my custom function which has liquid phase concentration and temp as parameters
def electrolyte_diffusivity(c,T):
    return (10**(-4.43-(54/(T-(229+5*c)))-0.22*c))*(10**-4)

// calling the custom function
// my concentration parameter will take the input from model.variables() as the liq phase concentration is constantly varying during the experiment.
param['Electrolyte diffusivity [m2.s-1]']=electrolyte_diffusivity(model.variables['Electrolyte concentration'],param['Ambient temperature [K]'])

// running this experiment gives the following error
experiment = pb.Experiment(
    [
        (
            "Charge at 1 C until 4.2 V",
            "Hold at 4.2 V until C/10",
            
            "Discharge at 1 C until 2.75 V",
            
        )
    ]
    *1
   
)
sim = pb.Simulation(model, experiment=experiment, parameter_values=param)
sim.solve(solver=pb.CasadiSolver(mode="safe"))
sim.plot(
    [
        'Terminal voltage [V]',
        'Discharge capacity [A.h]',
        
        
    ]
) 

//error --> TypeError: Parameter provided for 'Electrolyte diffusivity [m2.s-1]' is of the wrong type (should either be scalar-like or callable)

**Comment by valentinsulzer** [comment, marked as answer]
Just do `param['Electrolyte diffusivity [m2.s-1]']=electrolyte_diffusivity`
